File: Ruby/TestP2.rb
#encoding utf-8

require_relative 'Directions'
require_relative 'Dice'
require_relative 'GameCharacter'
require_relative 'Orientation'
require_relative 'Shield'
require_relative 'Weapon'
require_relative 'GameState'
require_relative 'Labyrinth'
require_relative 'Game'



class TestP2
    def main
        puts "Probando la practica 1:\n"

        puts "---------------Enum Directions-------------------"
        puts Irrgarten::Directions::LEFT
        puts Irrgarten::Directions::RIGHT
        puts Irrgarten::Directions::UP
        puts Irrgarten::Directions::DOWN

        puts "\n--------------Enum GameCharacter------------------"
        puts Irrgarten::GameCharacter::PLAYER
        puts Irrgarten::GameCharacter::MONSTER

        puts "\n--------------Enum Orientation-----------------"
        puts Irrgarten::Orientation::VERTICAL
        puts Irrgarten::Orientation::HORIZONTAL

        puts "\n--------------Creando instancias de clase-----------------"
        arma1 = Irrgarten::Weapon.new(3.2, 4)
        arma2 = Irrgarten::Weapon.new(1.1, 2)
        shield1 = Irrgarten::Shield.new(2.3, 3)
        shield2 = Irrgarten::Shield.new(1.2, 1)
        juego = Irrgarten::GameState.new("Garden of Bambam", "Mario:vivo, Miguel:vivo, Andrea:muerto", "Minotauro, Manticora",0, false,"Se ha creado el tablero de juego")

        puts "\n--------------Muestra de armas y escudos-----------------"
        puts "Arma 1: " + arma1.to_s
        puts "Arma 2: " + arma2.to_s
        puts "Escudo 1: " + shield1.to_s
        puts "Escudo 2: " + shield2.to_s
        
        puts "\nAtacando con arma 1, poder: " + arma1.attack.to_s
        puts "Compruebo usos: " + arma1.to_s
        puts "¿Tras el ataque se descarta? " + arma1.discard.to_s
        puts "\nProtegiendo con shield1, proteccion: " + shield1.protect.to_s
        puts "Compruebo usos: " + shield1.to_s
        puts "¿Tras proteger se descarta? " + shield1.discard.to_s

        puts "\n--------------Estado actual del juego-----------------"
        puts "labyrinth: " + juego.get_labyrinth
        puts "players: " + juego.get_players
        puts "monsters: " + juego.get_monsters
        puts "currentplayer: " + juego.get_current_player.to_s
        puts "winner: " + juego.get_winner.to_s
        puts "log: " + juego.get_log

        puts "\n----------------Prueba del dado-------------------"
        2.times do
            puts "\nRandom Position: #{Irrgarten::Dice.random_pos(100)}"
            puts "Who Starts: #{Irrgarten::Dice.who_starts(10)}"
            puts "Random Intelligence: #{Irrgarten::Dice.random_intelligence()}"
            puts "Random Strength: #{Irrgarten::Dice.random_strenght()}"
            puts "Resurrect Player: #{Irrgarten::Dice.resurrect_player()}"
            puts "Weapons Reward: #{Irrgarten::Dice.weapons_reward()}"
            puts "Shields Reward: #{Irrgarten::Dice.shields_reward()}"
            puts "Health Reward: #{Irrgarten::Dice.health_reward()}"
            puts "Weapon Power: #{Irrgarten::Dice.weapon_power()}"
            puts "Shield Power: #{Irrgarten::Dice.shield_power()}"
            puts "Numero de usos restante: #{Irrgarten::Dice.uses_left()}"
            puts "Intensidad: #{Irrgarten::Dice.intensity(5.0)}"
        end

        puts "\nProbando la practica 2:\n"
        puts "---------------Creando instancias de clase-------------------"
        unicorn = Irrgarten::Monster.new("Unicornio", Irrgarten::Dice.random_intelligence, Irrgarten::Dice.random_strenght)
        dragon = Irrgarten::Monster.new("Dragon", Irrgarten::Dice.random_intelligence, Irrgarten::Dice.random_strenght)
        minotaur = Irrgarten::Monster.new("Minotauro", Irrgarten::Dice.random_intelligence, Irrgarten::Dice.random_strenght)
        player1 = Irrgarten::Player.new('1', Irrgarten::Dice.random_intelligence, Irrgarten::Dice.random_strenght)
        player2 = Irrgarten::Player.new('2', Irrgarten::Dice.random_intelligence, Irrgarten::Dice.random_strenght)
        player3 = Irrgarten::Player.new('3', Irrgarten::Dice.random_intelligence, Irrgarten::Dice.random_strenght)
        labyrinth = Irrgarten::Labyrinth.new(3, 3, 2, 2)

        puts "\n---------------Muestra de instancias de clase-------------------"
        puts "Unicornio: " + unicorn.to_s
        puts "---------------------------------"
        puts "Dragon: " + dragon.to_s
        puts "---------------------------------"
        puts "Minotauro: " + minotaur.to_s
        puts "---------------------------------"
        puts "Player 1: " + player1.to_s
        puts "---------------------------------"
        puts "Player 2: " + player2.to_s
        puts "---------------------------------"
        puts "Player 3: " + player3.to_s
        puts "---------------Muestra de funciones------------------"
        player1.set_pos(1, 1)
        puts "Change position: " + player1.to_s
        puts "---------------------------------"
        puts "Player 1 dead: " + player1.dead.to_s
        puts "Player 1 resurrect: " + player1.resurrect.to_s
        puts "Player 1 attack: " + player1.attack.to_s
        puts "Unicorn dead: " + unicorn.dead.to_s
        puts "Unicorn attack: " + unicorn.attack.to_s

        puts "\n---------------Muestra de laberinto------------------"
        puts labyrinth.to_s
        puts "-----------------Add monster----------------"
        labyrinth.add_monster(2, 2, unicorn)
        puts labyrinth.to_s
        labyrinth.add_monster(1, 1, unicorn)
        puts labyrinth.to_s
        game = Irrgarten::Game.new(3)
        puts "-----------------Muestra de Game ----------------"
        estado = game.get_game_state
        puts estado.players.to_s
        puts "---------------------------------"
        puts estado.monsters.to_s
        puts "---------------------------------"
        puts estado.labyrinth.to_s

    end
end

test = TestP2.new
test.main

File: Java/Irrgarten/src/tests/TestP2.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package tests;

import irrgarten.Dice;
import irrgarten.Directions;
import irrgarten.GameCharacter;
import irrgarten.GameState;
import irrgarten.Orientation;
import irrgarten.Shield;
import irrgarten.Weapon;
import irrgarten.Labyrinth;
import irrgarten.Monster;
import irrgarten.Player;
import irrgarten.Game;

/**
 *
 * @author mariorl
 */
public class TestP2 {
    public static void main(String[] args) {
        System.out.println("Probando la practica 1:\n");
        
        System.out.println("---------------Enum Directions-------------------");
        System.out.println(Directions.DOWN);
        System.out.println(Directions.UP);
        System.out.println(Directions.LEFT);
        System.out.println(Directions.RIGHT);
    
        System.out.println("\n--------------Enum Orientation------------------");
        System.out.println(Orientation.HORIZONTAL);
        System.out.println(Orientation.VERTICAL);
        
        System.out.println("\n--------------Enum GameCharacter-----------------");
        System.out.println(GameCharacter.PLAYER);
        System.out.println(GameCharacter.MONSTER);
        
        System.out.println("\n--------------Creando instancias de clase-------------");
        Weapon arma1 = new Weapon(2.3f, 4);
        Weapon arma2 = new Weapon(1.7f, 1);
        Shield shield1 = new Shield(2.9f, 5);
        Shield shield2 = new Shield(4.1f, 2);
        GameState juego = new GameState("Garden of Bambam", "Mario:vivo, Miguel:vivo, Andrea:muerto", "Minotauro, Manticora",0, false,"Se ha creado el tablero de juego");
        
        System.out.println("\n--------------Muestra de armas y escudos----------------");
        System.out.println("arma1: " + arma1.toString());
        System.out.println("arma2: " + arma2.toString());
        System.out.println("shield1: " + shield1.toString());
        System.out.println("shield2: " + shield2.toString());
        
        System.out.println("\nAtaco con arma1: " + arma1.attack());
        System.out.println("arma1: " + arma1.toString());
        System.out.println("¿Tras el ataque se descarta?: " + arma1.discard());
        System.out.println("\nProtejo con shield1: " + shield1.protect());
        System.out.println("shield1: " + shield1.toString());
        System.out.println("¿Tras proteger se descarta?: " + shield1.discard());
        System.out.println("\nIntento de descartar arma2: " + arma2.discard());
        System.out.println("Intento de descartar shield2: " + arma2.discard());
        
        
        System.out.println("\n---------------Estado actual del juego------------------");
        System.out.println("labyrinth: " + juego.getLabyrinth());
        System.out.println("players: " + juego.getPlayers());
        System.out.println("monsters: " + juego.getMonsters());
        System.out.println("currentPlayer: " + juego.getCurrentPlayer());
        System.out.println("ganador: " + juego.getWinner());
        System.out.println("log: " + juego.getLog());
        
        System.out.println("\n---------------Prueba del dado---------------");
        for(int i = 0 ; i < 3 ; i++) {
            System.out.println("Posicion: " + Dice.randomPos(100));
            System.out.println("Inicia jugador: " + Dice.whoStarts(10));
            System.out.println("Con nivel de inteligencia: " + Dice.randomIntelligence());
            System.out.println("Con nivel de fuerza: " + Dice.randomStrenght());
            System.out.println("¿Resucita?: " + Dice.resurrectPlayer());
            System.out.println("Recompensa de armas: " + Dice.weaponsReward());
            System.out.println("Recompensa de escudos: " + Dice.shieldsReward());
            System.out.println("Recompensa de vida: " + Dice.healthReward());
            System.out.println("Poder del arma: " + Dice.weaponPower());
            System.out.println("Poder del escudo: " + Dice.shieldPower());
            System.out.println("Numero de usos restante: " + Dice.usesLeft());
            System.out.println("Intensidad: " + Dice.intensity(5.0f));
            System.out.println("\n");
        }
        
        System.out.println("Probando la practica 2:\n");
        System.out.println("--------------Creando instancias de clase-------------");
        Monster unicorn = new Monster("Unicorn",7.0f,2.0f);
        Monster minotaur = new Monster("Minotaur",4.0f,7.0f);
        Monster dragon = new Monster("Dragon",8.0f,9.0f);
        Player player0 = new Player('0',3.0f,7.0f);
        Player player1 = new Player('1',4.0f,6.0f);
        Player player2 = new Player('2',1.0f,9.0f);
        Labyrinth labyrinth = new Labyrinth(3,3,2,2);
        
        System.out.println("\n--------------Muestra de monstruos y jugadores----------------");
        System.out.println(unicorn.toString());
        System.out.println("------------------------------");
        System.out.println(minotaur.toString());
        System.out.println("------------------------------");
        System.out.println(dragon.toString());
        System.out.println("------------------------------");
        System.out.println(player0.toString());
        System.out.println("------------------------------");
        System.out.println(player1.toString());
        System.out.println("------------------------------");
        System.out.println(player2.toString());
        
        System.out.println("\n--------------Muestra de funciones----------------");
        player0.setPos(3, 4);
        System.out.println(player0.toString());
        System.out.println("------------------------------");
        System.out.println("getRow " + player0.getRow());
        System.out.println("getCol " + player0.getCol());
        System.out.println("Unicorn attack: " + unicorn.attack());
        System.out.println("Unicorn dead: " + unicorn.dead());
        System.out.println("Player0 attack: " + player0.attack());
        System.out.println("Player0 dead: " + player0.dead());
        
        System.out.println("\n--------------Muestra de laberinto----------------");
        System.out.println(labyrinth.toString());
        System.out.println("\n--------------Añadimos un monstruo----------------");
        labyrinth.addMonster(2, 2, minotaur);
        System.out.println(labyrinth.toString());
        labyrinth.addMonster(1, 1, unicorn);
        System.out.println(labyrinth.toString());
        Game game = new Game(3);
        System.out.println("\n--------------Muestra del juego----------------");
        GameState estado = game.getGameState();
        System.out.println(estado.getMonsters().toString());
        System.out.println(estado.getPlayers().toString());
        System.out.println(estado.getLabyrinth().toString());

        
    }
}
